plugins/poll/assets/javascripts/lib/chart-colors.js.es6

```diff
@@ -0,0 +1,71 @@
+export function getColors(count, palette) {
+  palette = palette || "cool";
+  let gradient;
+
+  switch (palette) {
+    case "cool":
+      gradient = {
+        0: [255, 255, 255, 1],
+        20: [220, 237, 200, 1],
+        45: [66, 179, 213, 1],
+        65: [26, 39, 62, 1],
+        100: [0, 0, 0, 1]
+      };
+      break;
+    case "warm":
+      gradient = {
+        0: [255, 255, 255, 1],
+        20: [254, 235, 101, 1],
+        45: [228, 82, 27, 1],
+        65: [77, 52, 47, 1],
+        100: [0, 0, 0, 1]
+      };
+      break;
+    case "neon":
+      gradient = {
+        0: [255, 255, 255, 1],
+        20: [255, 236, 179, 1],
+        45: [232, 82, 133, 1],
+        65: [106, 27, 154, 1],
+        100: [0, 0, 0, 1]
+      };
+      break;
+  }
+
+  //Get a sorted array of the gradient keys
+  let gradientKeys = Object.keys(gradient);
+  gradientKeys.sort(function(a, b) {
+    return +a - +b;
+  });
+
+  //Calculate colors
+  let colors = [];
+  for (let i = 0; i < count; i++) {
```

Comment:
This seems a little more complicated than it could be. The loop is O(n^3) now, looping through each count, gradient and 4 color values.

The gradients only increase, so there is no need to check all gradientKeys every iteration. Instead you could use a variable to represent the current value and one of the next value. Replace those values when your current index passes into the next value.